### PR TITLE
Phase2-hgx325A Update a good part of Validation/HGCalValidation on view of removal of several workflows

### PR DESCRIPTION
**Pull request by bsunanda**
#### PR description:

Update a good part of Validation/HGCalValidation on view of removal of several workflows

#### PR validation:

Use several test workflows

#### If this PR is a backport please specify the original PR and why you need to backport that PR. If this PR will be backported please specify to which release cycle the backport is meant for:

Nothing special

### CODE REVIEW

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by emanueleusai** [comment]
relval fetch timed out, let's retest

**Comment by emanueleusai** [comment]
please test

**Comment by emanueleusai** [comment]
type hgcal

**Review by rovere — CHANGES_REQUESTED** [review]
Ciao @bsunanda 
thanks a lot for having started the "cleaning" work. I added a couple of suggestions.

File: Validation/HGCalValidation/test/python/digiClient_cfg.py [hunk 1]
```diff
@@ -1,44 +1,75 @@
...
+    process = cms.Process('Client',Phase2C11M9)
+    process.load('Configuration.Geometry.GeometryExtended2026D92Reco_cff')
+    fileInput = 'file:DigiValD92.root'
+else:
+    from Configuration.Eras.Era_Phase2C11M9_cff import Phase2C11M9
+    process = cms.Process('Client',Phase2C11M9)
+    process.load('Configuration.Geometry.GeometryExtended2026D93Reco_cff')
+    fileInput = 'file:DigiValD93.root'
+
+print("Input file: ", fileInput)
```
Comment (rovere) [inline Validation/HGCalValidation/test/python/digiClient_cfg.py:45]: Wouldn't it be possible to use something like ```'{}'.format(options.geometry)``` to remove all the `ifs`?
That would also make the code more flexible for the future, no?
The last `else` is not checking any condition about which geometry has been supplied, right?

File: Validation/HGCalValidation/test/python/digiValidation_cfg.py [hunk 2]
```diff
@@ -1,7 +1,48 @@
...
+    process = cms.Process('testHGCalDigiLocal',Phase2C11M9)
+    process.load('Configuration.Geometry.GeometryExtended2026D92Reco_cff')
+    fileName = 'file:DigiValD92.root'
+else:
+    from Configuration.Eras.Era_Phase2C11M9_cff import Phase2C11M9
+    process = cms.Process('testHGCalDigiLocal',Phase2C11M9)
+    process.load('Configuration.Geometry.GeometryExtended2026D93Reco_cff')
+    fileName = 'file:DigiValD93.root'
+
+print("Output file: ", fileName)
```
Comment (rovere) [inline Validation/HGCalValidation/test/python/digiValidation_cfg.py:45]: The same comment applies here.

File: Validation/HGCalValidation/test/python/protoHGCalSimWatcher_cfg.py [hunk 3]
```diff
@@ -26,62 +26,27 @@
...
-    process.load('Configuration.Geometry.GeometryExtended2026D77_cff')
-    process.load('Configuration.Geometry.GeometryExtended2026D77Reco_cff')
-    fileCheck = 'testHGCalSimWatcherV14.root'
-    runMode = 0
+    process.load('Configuration.Geometry.GeometryExtended2026D93Reco_cff')
+    fileCheck = 'testHGCalSimWatcherV17N.root'
+    runMode = 1
+
+print("Output file: ", fileCheck)
+print("Run Mode:    ", runMode)
```
Comment (rovere) [inline Validation/HGCalValidation/test/python/protoHGCalSimWatcher_cfg.py:50]: The same comment applies here as well. There's here the complication of handling correctly the `fileCheck` variable. Maybe the previous naming convention is be a better choice that could be applied here as well?

`runMode` is always set to 1: is it really needed?

File: Validation/HGCalValidation/test/python/protoSimValid_cfg.py [hunk 4]
```diff
@@ -32,78 +32,32 @@
...
-    process.load('Configuration.Geometry.GeometryExtended2026D77Reco_cff')
+    process.load('Configuration.Geometry.GeometryExtended2026D93Reco_cff')
     if (options.type == "hgcalSiliconValidation"):
-        fileName = 'hgcSilValidD77.root'
+        fileName = 'hgcSilValidD93.root'
     else:
-        fileName = 'hgcBHValidD77.root'
+        fileName = 'hgcBHValidD93.root'
+
+print("Output file: ", fileName)
```
Comment (rovere) [inline Validation/HGCalValidation/test/python/protoSimValid_cfg.py:60]: Same comment

File: Validation/HGCalValidation/test/python/recHitClient_cfg.py [hunk 5]
```diff
@@ -1,43 +1,77 @@
...
+    process = cms.Process('Client',Phase2C11M9)
+    process.load('Configuration.Geometry.GeometryExtended2026D92Reco_cff')
+    fileInput = 'file:RecHitValD92.root'
+else:
+    from Configuration.Eras.Era_Phase2C11M9_cff import Phase2C11M9
+    process = cms.Process('Client',Phase2C11M9)
+    process.load('Configuration.Geometry.GeometryExtended2026D93Reco_cff')
+    fileInput = 'file:RecHitValD93.root'
+
+print("Input file: ", fileInput)
```
Comment (rovere) [inline Validation/HGCalValidation/test/python/recHitClient_cfg.py:46]: Same comment here

File: Validation/HGCalValidation/test/python/rechitValidation_cfg.py [hunk 6]
```diff
@@ -1,7 +1,52 @@
...
+    outFile = 'file:step3D92.root'
+else:
+    from Configuration.Eras.Era_Phase2C11M9_cff import Phase2C11M9
+    process = cms.Process('testHGCalRecHitLocal',Phase2C11M9)
+    process.load('Configuration.Geometry.GeometryExtended2026D93Reco_cff')
+    fileName = 'file:RecHitValD93.root'
+    outFile = 'file:step3D93.root'
+
+print("Reco Output file:       ", outFile)
+print("Validation Output file: ", fileName)
```
Comment (rovere) [inline Validation/HGCalValidation/test/python/rechitValidation_cfg.py:50]: Same comment here

File: Validation/HGCalValidation/test/python/runHFNoseDigiStudy_cfg.py [hunk 7]
```diff
@@ -19,7 +18,7 @@
     process.MessageLogger.HGCalGeom=dict()
 
 process.source = cms.Source("PoolSource",
-                            fileNames = cms.untracked.vstring('file:step2.root')
+                            fileNames = cms.untracked.vstring('file:step2D94.root')
                             )
```
Comment (rovere) [inline Validation/HGCalValidation/test/python/runHFNoseDigiStudy_cfg.py:22]: Would it be useful to add the same flexibility of handling the geometry via a command line parameter?
Probab;y `D94` won't be the last geometry we will use for `HFNose`.

File: Validation/HGCalValidation/test/python/runHFNoseRecHitStudy_cfg.py [hunk 8]
```diff
@@ -20,7 +19,7 @@
     process.MessageLogger.HGCalValidation=dict()
 
 process.source = cms.Source("PoolSource",
-                            fileNames = cms.untracked.vstring('file:step3.root')
+                            fileNames = cms.untracked.vstring('file:step3D94.root')
                             )
```
Comment (rovere) [inline Validation/HGCalValidation/test/python/runHFNoseRecHitStudy_cfg.py:23]: Same comment here

File: Validation/HGCalValidation/test/python/runHFNoseSimHitStudy_cfg.py [hunk 9]
```diff
@@ -18,7 +18,7 @@
     process.MessageLogger.HGCalValidation=dict()
 
 process.source = cms.Source("PoolSource",
-                            fileNames = cms.untracked.vstring('file:step1.root')
+                            fileNames = cms.untracked.vstring('file:step1D94.root')
```
Comment (rovere) [inline Validation/HGCalValidation/test/python/runHFNoseSimHitStudy_cfg.py:21]: Same comment here

File: Validation/HGCalValidation/test/python/runHGCHitAnalyzer_cfg.py [hunk 10]
```diff
@@ -1,24 +1,52 @@
...
+    outFile = 'relValTTbarD92.root'
+else:
+    from Configuration.Eras.Era_Phase2C11M9_cff import Phase2C11M9
+    process = cms.Process('HGCHitAnalysis',Phase2C11M9)
+    process.load('Configuration.Geometry.GeometryExtended2026D93Reco_cff')
+    inFile = 'file:step3D93.root'
+    outFile = 'relValTTbarD93.root'
+
+print("Input file: ", inFile)
+print("Output file: ", outFile)
```
Comment (rovere) [inline Validation/HGCalValidation/test/python/runHGCHitAnalyzer_cfg.py:50]: Same comment here

File: Validation/HGCalValidation/test/python/simHitClient_cfg.py [hunk 11]
```diff
@@ -1,36 +1,72 @@
...
+    process = cms.Process('Client',Phase2C11M9)
+    process.load('Configuration.Geometry.GeometryExtended2026D92Reco_cff')
+    fileInput = 'file:SimHitValD92.root'
+else:
+    from Configuration.Eras.Era_Phase2C11M9_cff import Phase2C11M9
+    process = cms.Process('Client',Phase2C11M9)
+    process.load('Configuration.Geometry.GeometryExtended2026D93Reco_cff')
+    fileInput = 'file:SimHitValD93.root'
+
+print("Input file: ", fileInput)
```
Comment (rovere) [inline Validation/HGCalValidation/test/python/simHitClient_cfg.py:45]: Same comment here

File: Validation/HGCalValidation/test/python/simHitValidation_cfg.py [hunk 12]
```diff
@@ -1,90 +1,162 @@
...
+    outFile = 'file:step1D92.root'
+else:
+    from Configuration.Eras.Era_Phase2C11M9_cff import Phase2C11M9
+    process = cms.Process('testHGCalSimHitLocal',Phase2C11M9)
+    process.load('Configuration.Geometry.GeometryExtended2026D93Reco_cff')
+    fileName = 'file:SimHitValD93.root'
+    outFile = 'file:step1D93.root'
+
+print("SIM Output file:        ", outFile)
+print("Validation Output file: ", fileName)
```
Comment (rovere) [inline Validation/HGCalValidation/test/python/simHitValidation_cfg.py:50]: Same

File: Validation/HGCalValidation/test/python/testHFNoseSimHitStudy_cfg.py [hunk 13]
```diff
@@ -80,22 +78,21 @@
         dataTier = cms.untracked.string('GEN-SIM'),
         filterName = cms.untracked.string('')
     ),
-    fileName = cms.untracked.string('file:simevent.root'),
+    fileName = cms.untracked.string('file:step1D94.root'),
```
Comment (rovere) [inline Validation/HGCalValidation/test/python/testHFNoseSimHitStudy_cfg.py:81]: Same

File: Validation/HGCalValidation/test/python/testHGCalBHValid_cfg.py [hunk 14]
```diff
@@ -26,30 +26,21 @@
...
 else:
-    from Configuration.Eras.Era_Phase2C11_cff import Phase2C11
-    process = cms.Process('HGCGeomAnalysis',Phase2C11)
-    process.load('Configuration.Geometry.GeometryExtended2026D77_cff')
-    process.load('Configuration.Geometry.GeometryExtended2026D77Reco_cff')
-    fileName = 'hgcBHValidD77.root'
+    from Configuration.Eras.Era_Phase2C11M9_cff import Phase2C11M9
+    process = cms.Process('HGCGeomAnalysis',Phase2C11M9)
+    process.load('Configuration.Geometry.GeometryExtended2026D93Reco_cff')
+    fileName = 'hgcBHValidD93.root'
```
Comment (rovere) [inline Validation/HGCalValidation/test/python/testHGCalBHValid_cfg.py:43]: Same

**Review by rovere — CHANGES_REQUESTED** [review]

File: Validation/HGCalValidation/test/python/digiClient_cfg.py (hunk 1 above)
Comment (bsunanda) [reply Validation/HGCalValidation/test/python/digiClient_cfg.py:45]: Please approve this PR. I am working on using an array and implement it later once tested

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by emanueleusai** [comment]
+1

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by emanueleusai** [comment]
+1

- resign

**Comment by perrotta** [comment]
+1

**Pull request merged**